diff --git a/fs/proc/version.c b/fs/proc/version.c
index d2154eb..21ccc9b 100644
--- a/fs/proc/version.c
+++ b/fs/proc/version.c
@@ -4,10 +4,16 @@
 #include <linux/proc_fs.h>
 #include <linux/seq_file.h>
 #include <linux/utsname.h>
+#include <generated/compile.h>
+
+const char lp_linux_proc_banner[] =
+	"%s version %s-livepatch"
+	" (" LINUX_COMPILE_BY "@" LINUX_COMPILE_HOST ")"
+	" (" LINUX_COMPILER ") %s\n";
 
 static int version_proc_show(struct seq_file *m, void *v)
 {
-	seq_printf(m, linux_proc_banner,
+	seq_printf(m, lp_linux_proc_banner,
 		utsname()->sysname,
 		utsname()->release,
 		utsname()->version);
-- 
2.1.0

